docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -54,27 +71,43 @@ VSOutput Main(VSInput input)
 }
 ```
 
-First off, we define 2 types, `VSInput` and `VSOutput` which represent the vertex shader's input and output.
+First off, we define 2 types, `VSInput` and `VSOutput` which represent the 
+vertex shader's input and output.
+
+The input is 2, `float3` (vector of 3 float components), the first is the 
+"position" field, which are coordinates ranging from [-1.0, 1.0], if the values 
+are outside this range they are clipped, and we won't see them on screen.
 
-The input is 2, `float3` (vector of 3 float components), the first is the "position" field, which are coordinates ranging from [-1.0, 1.0], if the values are outside this range they are clipped, and we won't see them on screen.
+The second is the "color" field, which we also pass as the output of this stage 
+onto the pixel shader.
 
-The second is the "color" field, which we also pass as the output of this stage onto the pixel shader.
+Notice how all our fields have a colon and some identifier attached to them, 
+these are "semantics". Semantics that are preceded by `SV` are called 
+"system-value semantics" and their meaning and usage is defined by D3D11. 
+`SV_Position` for example means that the field `position` will be used by 
+D3D11 as the actual output of the vertex shader.
 
-Notice how all our fields have a colon and some identifier attached to them, these are "semantics".
-Semantics that are preceded by `SV` are called "system-value semantics" and their meaning and usage is defined by D3D11. `SV_Position` for example means that the field `position` will be used by D3D11 as the actual output of the vertex shader.
-Everything else are "user defined semantics" and their naming is up to us. These are used to pass data between shader stages.
+Everything else are "user defined semantics" and their naming is up to us. 
+These are used to pass data between shader stages.
 
-Then we have our `VSOutput`, which has our vertices in the first field `position` and our color in the second field `color`.
+Then we have our `VSOutput`, which has our vertices in the first field `position` 
+and our color in the second field `color`.
 
-Finally, we have our main function, which takes in a single parameter which is our input in the form of `VSInput`, and returns our output in the form of a `VSOutput`. Since we don't do any processing, we simply make a new instance of `VSOutput`, initialize it all to 0 and forward our input position and color to the output.
+Finally, we have our main function, which takes in a single parameter which is 
+our input in the form of `VSInput`, and returns our output in the form of a `VSOutput`.
+ Since we don't do any processing, we simply make a new instance of `VSOutput`, 
+ initialize it all to 0 and forward our input position and color to the output.
```

Comment:
splitting hairs, but having only structures in/output isn't required, you can mention each input seperately and directly declare output as well:
eg;
`float4 main(float3 inPos : POSITION, float2 inUV: TEXCOORD) : SV_Position`

Not sure we want to mention this, just adding information here

Reply:
Yep this is possible, should definitly be mentioned, but it seems kinda convention that you use structs - its more readable IMHO (mileage my wary)

Reply:
I think structs are a better convention as well- plus if we explain it with an example of separately it may be very confusing later on with how we explain things using a frame of reference